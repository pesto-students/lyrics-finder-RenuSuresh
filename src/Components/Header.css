.header {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 50vh;
}
.header__name {
  font-family: "Oleo Script", sans-serif;
  font-size: 50px;
  color: aliceblue;
  display: flex;
  justify-content: center;
  align-items: center;
}
.header__search {
  display: flex;
  justify-content: center;
}
.header__input__search {
  width: 130px;
  box-sizing: border-box;
  border-radius: 4px;
  font-size: 16px;
  background-color: white;
  background-position: 10px 10px;
  background-repeat: no-repeat;
  padding: 12px 20px 12px 40px;
  -webkit-transition: width 0.4s ease-in-out;
  transition: width 0.4s ease-in-out;
  float: right;
  outline: none;
}
.header__input__search:focus {
  width: 100%;
  outline: none;
}
