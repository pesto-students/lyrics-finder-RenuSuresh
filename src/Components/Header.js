import React from "react";
import "./Header.css";
function Header() {
  return (
    <div className="header">
      <p className="header__name">Lyrics Finder</p>
      <div className="header__search">
        <input className="header__input__search" />
        <button>Search</button>
      </div>
    </div>
  );
}

export default Header;
