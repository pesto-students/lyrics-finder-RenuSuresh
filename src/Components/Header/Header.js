import React, { useState } from "react";
import "./Header.css";
import Table from "../Table/Table";

function Header({ toggleHome }) {
  const [search, setSearch] = useState("");
  const [isSearch, setIsSearch] = useState(false);
  const [isError, setIsError] = useState("");
  const getSearch = () => {
    if (search) {
      setIsSearch(true);
      toggleHome(false);
    } else {
      toggleHome(false);
      setIsError(true);
    }
  };
  const hideTable = () => {
    setIsSearch(false);
    toggleHome(true);
    setSearch("");
  };
  return (
    <div>
      <div className="header">
        <p className="header__name font-effect-shadow-multiple">
          Lyrics Finder
        </p>
        <div style={{ display: "flex", flexDirection: "column" }}>
          <div className="header__search">
            <input
              className="header__input__search"
              placeholder="Search by Artist/ Song"
              name="search"
              onChange={(e) => {
                setSearch(e.target.value);
                setIsError(false);
              }}
              value={search}
              data-testid="header-search"
            />

            <button
              className="header__searchbtn"
              onClick={getSearch}
              data-testid="search-btn"
            >
              <i className="fa fa-search"></i>
            </button>
          </div>
          {isError && (
            <p
              style={{
                color: "white",
                fontSize: "12px",
                marginTop: "1%",
              }}
              data-testid="error-text"
            >
              * This field is required.
            </p>
          )}
        </div>
      </div>

      {isSearch && <Table musicType={search} hideTable={hideTable} />}
    </div>
  );
}

export default Header;
