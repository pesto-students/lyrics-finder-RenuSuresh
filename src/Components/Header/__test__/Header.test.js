import { configure, EnzymeAdapter } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { shallow } from "enzyme";
import Header from "../Header";
import Enzyme from "enzyme";
import { render, fireEvent, getByTestId } from "@testing-library/react";
import toJson from "enzyme-to-json";
configure({ adapter: new Adapter() });
const { mount } = Enzyme;
describe("Header", () => {
  test("snapshot <Header/>", () => {
    const headerTree = mount(<Header />);
    expect(toJson(headerTree)).toMatchSnapshot();
  });
  test("header testing", () => {
    const headerWrapper = mount(<Header />);
    const headerText = headerWrapper.find(".header__name");
    expect(headerText.text()).toBe("Lyrics Finder");
  });

  describe("behaviour", () => {
    test("pass props of toggleHome", () => {
      const headerWrapper = mount(<Header />);
      const table = headerWrapper;
    });
  });
  test("Initially input field to be empty", () => {
    const { container } = render(<Header />);
    const inputValue = getByTestId(container, "header-search");
    expect(inputValue.textContent).toBe("");
  });
  test("Input field on change", () => {
    const { container } = render(<Header />);
    const inputValue = getByTestId(container, "header-search");
    const newSearch = "coldplay";
    fireEvent.change(inputValue, { target: { value: newSearch } });
    expect(inputValue.value).toBe(newSearch);
  });
  test("submit button", () => {
    const { container } = render(<Header />);
    const submitBtn = getByTestId(container, "search-btn");
    expect(submitBtn.value).toBe("");
  });
  test("if submit without entering input to be empty", () => {
    const mock = jest.fn();
    const { container } = render(<Header toggleHome={mock} />);
    const submitBtn = getByTestId(container, "search-btn");

    const inputValue = getByTestId(container, "header-search");
    fireEvent.click(submitBtn);

    expect(inputValue.value).toBe("");
  });
  test("if submit without any input show error", () => {
    const mock = jest.fn();
    const { container } = render(<Header toggleHome={mock} />);
    const submitBtn = getByTestId(container, "search-btn");

    const inputValue = getByTestId(container, "header-search");
    fireEvent.click(submitBtn);

    const errorText = getByTestId(container, "error-text");
    const errorValue = "* This field is required.";
    expect(errorText.textContent).toBe(errorValue);
  });
});
