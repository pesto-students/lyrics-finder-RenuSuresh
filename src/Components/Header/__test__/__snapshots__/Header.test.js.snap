// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot <Header/> 1`] = `
<Header>
  <div>
    <div
      className="header"
    >
      <p
        className="header__name font-effect-shadow-multiple"
      >
        Lyrics Finder
      </p>
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <div
          className="header__search"
        >
          <input
            className="header__input__search"
            data-testid="header-search"
            name="search"
            onChange={[Function]}
            placeholder="Search by Artist/ Song"
            value=""
          />
          <button
            className="header__searchbtn"
            data-testid="search-btn"
            onClick={[Function]}
          >
            <i
              className="fa fa-search"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</Header>
`;
